CODEBASE_INTRO.md

```diff
@@ -0,0 +1,170 @@
+
+- **Initial effect setup**: The first visual effect is set up and started. This ensures that as soon as the system is ready, there's immediate visual feedback on the LED display.
+
+- **Loading the full effect set**: The complete effects list is loaded, either from a JSON file stored on the device's SPIFFS partition or the default set. The default set is determined by the effect factories that are added in the `LoadEffectFactories()` function in [effects.cpp](./src/effects.cpp).
+
+### Event loop
+
+Once all the initializations are complete, the application enters its main event loop. _**Note** that the event loop is not an actual function or separate thread._ Instead, it's implemented by the combination of threads that take in and process different types of input. In that, it continuously listens for events (like user inputs, network commands, or audio signals) and updates the LED animations accordingly.
+
+### Main drawing loop
+
+The `EffectManager` and the drawing loop work together to render the effects that have been enabled within the application.
+
+- **`EffectManager`**: The `EffectManager` is a class that is present in the [effectmanager.h](./include/effectmanager.h) and, for a few methods and support functions, [effects.cpp](./src/effects.cpp) files. It  manages the LED effects by holding the list of effect objects that represent individual effects that can be shown on the LED display.
+  The `EffectManager` has a method named `Update()`, which is responsible for drawing the current effect on the LED display. This method is called within the main draw loop to render the effect.
+
+- **Drawing loop**: The drawing loop is present in the [drawing.cpp](./src/drawing.cpp) file. It's a continuous loop that updates the LED display with the current effect.
+Within this loop, the `EffectManager`'s `Update()` method is called, which in turn calls the `Draw()` method of the current effect (object). This ensures that the effect is rendered on the LED display.
+
+- **Effect classes**: Each of the effect classes, as present in the various files in the [include/effects](./include/effects/) subfolders, represents a single effect that can be applied to the LED display. All effect classes, including those targeting LED matrices, eventually derive from the `LEDStripEffect` class as found in [ledstripeffect.h](./include/ledstripeffect.h) and [ledstripeffect.cpp](./src/ledstripeffect.cpp).
+Each effect has its own implementation of the `Draw()` method, which defines how the effect should be rendered on the LED display. This method is called by the `EffectManager` when it's time to render the effect.
+
+## Memory management and performance
+
+Given the real-time nature of LED displays, performance is paramount. The codebase makes extensive use of dynamic memory allocation, with a preference for PSRAM memory when available.

```

Comment:
I think this may also be a good point to touch upon in a more detailed discussion on how to code. At the same time, I'll be honest that I would struggle myself with how to phrase the actual advice to give. Beyond "if things seem to behave oddly with PSRAM, try regular RAM and see if the oddness disappears", that is.